src/GafferDispatchModule/DispatcherBinding.cpp

```diff
@@ -294,6 +294,28 @@ struct PreDispatchSlotCaller
 	}
 };
 
+struct DispatchSlotCaller
+{
+	boost::signals::detail::unusable operator()( boost::python::object slot, const Dispatcher *d, const std::vector<TaskNodePtr> &nodes )
+	{
+		try
+		{
+			list nodeList;
+			for( std::vector<TaskNodePtr>::const_iterator nIt = nodes.begin(); nIt != nodes.end(); nIt++ )
+			{
+				nodeList.append( *nIt );
+			}
+			DispatcherPtr dd = const_cast<Dispatcher*>(d);
+			slot( dd, nodeList );
+		}
+		catch( const error_already_set &e )
+		{
+			PyErr_PrintEx( 0 ); // clears the error status
```

Comment:
We should be using `ExceptionAlgo::translatePythonException()` here and then relying on a CatchingSignalCombiner on the C++ side. This `PyErr_PrintEx()` approach is a poor man's version of that which predates the exception translation code. Now would be a good time to update the other uses of `PyErr_PrintEx()` in this file too I reckon.

Reply:
Addressed in 514c4e7446332a44b9670bf96b233107c8a4bdc9